src/js/api/brand.js

```diff
@@ -759,6 +769,58 @@ ripe.Ripe.prototype._getMeshOptions = function(options = {}) {
     });
 };
 
+/**
+ * @ignore
+ */
+ripe.Ripe.prototype._getInitials3dOptions = function(options = {}) {
+    const brand = options.brand === undefined ? this.brand : options.brand;
+    const version = options.version === undefined ? this.version : options.version;
+    const variant = options.variant === undefined ? this.variant : options.variant;
+    const url = `${this.url}brands/${brand}`;
+    const params = {};
+    if (version !== undefined && version !== null) {
+        params.version = version;
+    }
+    if (variant !== undefined && variant !== null) {
+        params.variant = variant;
+    }
+
+    let baseTexture = null;
+    let displacementTexture = null;
+    let metallicTexture = null;
+    let normalTexture = null;
+    let roughnessTexture = null;
+    const config = this.loadedConfig || {};
+    const profiles = (config.initials && config.initials.profiles) || [];
+    const $profiles = (config.initials && config.initials.$profiles) || {};
+    profiles.forEach(profile => {
+        const values = $profiles[profile] || {};
+        const _3d = values["3d"] || {};
+        baseTexture = _3d.base_texture;
+        displacementTexture = _3d.displacement_texture;
+        metallicTexture = _3d.metallic_texture;
+        normalTexture = _3d.normal_texture;
+        roughnessTexture = _3d.roughness_texture;
+    });

```

Comment:
we should also take into account the aliases in the spec

Back then I implemented something like this

```javascript
        const modelProfileSpec = config.initials.$profiles || {};
        const profileAliases = config.initials.$alias || {};
        const enrichedProfiles = Object.entries(profileAliases).reduce((p, [key, value]) => {
            return profiles.includes(key) ? [...p, ...value] : p;
        }, profiles);
        const modelBaseSpec = modelProfileSpec.base || {};
        const mergedSpec = enrichedProfiles.reduce((p, profile) => {
            return { ...p, ...modelProfileSpec[profile] };
        }, modelBaseSpec);
        console.log("mergedSpec", mergedSpec);
```

This gets the complete spec (mergedSpec) considering the current profile and aliases.

Reply:
I have a question,
Aren't the `$alias` just that, alias to the profiles names?

What's the idea behind the change, because at it is now what it does is just apply the profiles in their respecting the it's order

Reply:
The idea is if you have an alias called "report" in the `config.initials.profiles` that is an alias to apply 3 different profiles this will not work as expected - it will try to search for the properties of profile "report" when it should, instead, search for the properties of each profile of the alias "report".

The idea is adding the profiles that are inside a certain alias and **only after** having the (what I called `enrichedProfiles`) should you do what you're doing in your code at `profiles.forEach(...`.

Reply:
Hmm, what about the profiles order? Doesn't that mean that as it is now, we don't guarantee the profiles order?
I'm asking because `$alias`  is an object

Reply:
I'm not sure about that but I found a build for it https://github.com/ripe-tech/builds-static/blob/master/config.static/dior_tech/jadore/spec.json - maybe try to play with it and find out what's our current default behavior

Reply:
Took a deeper dive on how the whole profile thing worked and improved how the values are obtained. Should be good now, take a look